.github/workflows/build.yml

```diff
@@ -238,6 +244,9 @@ jobs:
            mkdir -p ~/private_keys/
            echo '${{ secrets.api_key }}' > ~/private_keys/AuthKey_${{ secrets.api_key_id }}.p8
 
+       - name: Compile components
```

Comment:
What is the reason we compile components 3x?
Can't we cache the result of 1st complication and reuse in other 2 places?

Reply:
Roughly the same reason we compile the app in those 3 places too: the are being used as a build step prior to the app build step, so whenever the app gets built the components should be built as well.

It should be possible to cache the results of the compilation, especially since the component compilation is simpler than the app compilation. But caching brings its own set of problems, like determining the cache invalidation policy (yarn.lock+all ts files for now, maybe different later), and I can't imagine it will bring any benefit since compiling the components is faster than restoring cache+storing cache for now.

So I don't really see a reason to use cache at the moment.

Reply:
They aren't served anywhere now, no. I think they can be built with `yarn build-storybook` in this fashion. Afterwards they would need to be deployed.

For contributors to show them they'd also have to somehow deploy them to github pages though. How was that done before?

Reply:
Another note: athensresearch.github.io is essentially a singleton for the master branch deployment, we can't deploy there until rtc is merged into master.

Reply:
> For contributors to show them they'd also have to somehow deploy them to github pages though. How was that done before?

https://handbook.athensresearch.org/community/get-involved/setup#deploying-athens-and-devcards

> Another note: athensresearch.github.io is essentially a singleton for the master branch deployment, we can't deploy there until rtc is merged into master.

That's a good point. Right now, Github pages are only updated when main is deployed via a tagged release. 

In the past, I setup Actions so that contributors would update their GitHub pages whenever they pushed to any branch, not just when they pushed to their main. The problem here was that if they pushed to another branch, it would update. I even dynamically included their Git hash and linked to the GitHub commit in their GitHub pages Athens build. Actually, none of this worked out that well, and the easiest ended up being that I just added their fork as a remote and built it locally every time I was reviewing their PR. This should probably still be the workflow going forward for PR reviews.

Reply:
The problem with GH pages is it would only be able to serve one branch at a time. I feel like a service like Vercel could have a deploy for an arbitrary number of git hashes. This is too much overhead for most contributors, but I could seem it being more valuable when we have several people iterating on frontend code.

Reply:
WRT https://github.com/athensresearch/athens/pull/1564#discussion_r698394605, talked to Alex on standup and he prefer the faster build than the caching for now.